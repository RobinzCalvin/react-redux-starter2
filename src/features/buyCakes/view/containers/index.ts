export { default as CakePreviewList } from './CakePreviewList/CakePreviewList';
