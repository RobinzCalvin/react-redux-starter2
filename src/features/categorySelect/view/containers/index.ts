export { default as CategorySelect } from './CategorySelect/CategorySelect';
