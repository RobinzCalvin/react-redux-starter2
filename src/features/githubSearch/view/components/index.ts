export { default as UserAvatarsWall } from './UserAvatarsWall/UserAvatarsWall';
export { default as SearchSettingsDialog } from './SearchSettingsDialog/SearchSettingsDialog';
