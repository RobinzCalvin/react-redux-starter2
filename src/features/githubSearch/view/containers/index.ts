export { default as UserSearchForm } from './UserSearchForm/UserSearchForm';
export { default as UserSearchResults } from './UserSearchResults/UserSearchResults';
