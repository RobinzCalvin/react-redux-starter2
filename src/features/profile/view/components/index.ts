export { default as ProfileAvatar } from './ProfileAvatar/ProfileAvatar';
