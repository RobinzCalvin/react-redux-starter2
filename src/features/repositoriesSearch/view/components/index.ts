export { default as RepositoryPreview } from './RepositoryPreview/RepositoryPreview';
