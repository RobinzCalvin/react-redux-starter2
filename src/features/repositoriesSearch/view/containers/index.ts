export { default as RepositoriesSearchForm } from './RepositoriesSearchForm/RepositoriesSearchForm';
