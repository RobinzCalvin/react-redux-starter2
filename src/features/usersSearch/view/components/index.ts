export { default as UserAvatarsWall } from './UserAvatarsWall/UserAvatarsWall';
export { default as UserAttribute } from './UserAttribute/UserAttribute';
