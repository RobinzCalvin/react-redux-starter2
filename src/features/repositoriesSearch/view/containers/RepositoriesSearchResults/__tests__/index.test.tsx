import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import { makeMockContainer, repository } from 'shared/mocks';
import { PaginationControls } from 'shared/view/components';

import { RepositoryPreview } from '../../../components';
import {
  RepositoriesSearchResults, IRepositoriesSearchResultsProps, IRepositoriesSearchResultsState,
} from '../RepositoriesSearchResults';

const props: IRepositoriesSearchResultsProps = {
  searchOptions: {
    searchString: 'search',
  },
  repositories: Array(10).fill(repository),
  paginationState: {
    page: 1,
    totalPages: 1,
  },
  searchRepositories: jest.fn(),
  UserDetails: makeMockContainer('UserDetails'),
};

describe('(features/repositoriesSearch) RepositoriesSearchResults container', () => {
  let component: ShallowWrapper<IRepositoriesSearchResultsProps, IRepositoriesSearchResultsState>;
  beforeEach(() => component = shallow(<RepositoriesSearchResults {...props} />));

  it('should render all found repositories', () => {
    const renderedRepos = component.find('.repositories-search-results__repository-preview');
    expect(renderedRepos.length).toBe(props.repositories.length);
  });

  it('should call searchRepositories with search options and page number on PaginationControls page request', () => {
    const page = 1;
    const { searchOptions, searchRepositories } = props;
    component.find(PaginationControls).prop('onPageRequest')(page);
    expect(searchRepositories).toHaveBeenCalledWith({ searchOptions, page });
  });

  it('should render user details after RepositoryPreview owner click', () => {
    const { UserDetails } = props;
    expect(component.find(UserDetails).length).toBe(0);

    const { username } = props.repositories[0].owner;
    const repositoryPreview = component.find(RepositoryPreview).at(0);
    repositoryPreview.prop('onOwnerClick')(username);
    expect(component.find(UserDetails).length).toBe(1);
  });

  it('should hide user details after UserDetails onClose call', () => {
    const { UserDetails } = props;
    component.setState({ displayedRepositoryOwner: 'owner' });
    expect(component.find(UserDetails).length).toBe(1);

    const userDetails = component.find(UserDetails);
    userDetails.prop('onClose')();
    expect(component.find(UserDetails).length).toBe(0);
  });
});
