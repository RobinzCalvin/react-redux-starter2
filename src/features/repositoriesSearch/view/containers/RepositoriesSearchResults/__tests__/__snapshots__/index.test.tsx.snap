// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepositoriesSearchResults component should match snapshot 1`] = `
<div
  className="repositories-search-results"
>
  <div
    className={
      Object {
        "has": [Function],
        "is": [Function],
        "mix": [Function],
        "split": [Function],
        "state": [Function],
        "toString": [Function],
      }
    }
    key="1"
  >
    <RepositoryPreview
      onOwnerClick={[Function]}
      repository={
        Object {
          "description": "description",
          "forksNumber": 1,
          "htmlURL": "http://the-url.com",
          "id": 1,
          "language": "en",
          "name": "name",
          "openIssuesNumber": 1,
          "owner": Object {
            "avatarURL": "http://the-url.com",
            "htmlURL": "http://the-url.com",
            "id": 1,
            "username": "username",
          },
          "starsNumber": 1,
          "updatedAt": "05/05/2005",
        }
      }
    />
  </div>
  <div
    className={
      Object {
        "has": [Function],
        "is": [Function],
        "mix": [Function],
        "split": [Function],
        "state": [Function],
        "toString": [Function],
      }
    }
    key="1"
  >
    <RepositoryPreview
      onOwnerClick={[Function]}
      repository={
        Object {
          "description": "description",
          "forksNumber": 1,
          "htmlURL": "http://the-url.com",
          "id": 1,
          "language": "en",
          "name": "name",
          "openIssuesNumber": 1,
          "owner": Object {
            "avatarURL": "http://the-url.com",
            "htmlURL": "http://the-url.com",
            "id": 1,
            "username": "username",
          },
          "starsNumber": 1,
          "updatedAt": "05/05/2005",
        }
      }
    />
  </div>
  <div
    className={
      Object {
        "has": [Function],
        "is": [Function],
        "mix": [Function],
        "split": [Function],
        "state": [Function],
        "toString": [Function],
      }
    }
    key="1"
  >
    <RepositoryPreview
      onOwnerClick={[Function]}
      repository={
        Object {
          "description": "description",
          "forksNumber": 1,
          "htmlURL": "http://the-url.com",
          "id": 1,
          "language": "en",
          "name": "name",
          "openIssuesNumber": 1,
          "owner": Object {
            "avatarURL": "http://the-url.com",
            "htmlURL": "http://the-url.com",
            "id": 1,
            "username": "username",
          },
          "starsNumber": 1,
          "updatedAt": "05/05/2005",
        }
      }
    />
  </div>
  <div
    className={
      Object {
        "has": [Function],
        "is": [Function],
        "mix": [Function],
        "split": [Function],
        "state": [Function],
        "toString": [Function],
      }
    }
    key="1"
  >
    <RepositoryPreview
      onOwnerClick={[Function]}
      repository={
        Object {
          "description": "description",
          "forksNumber": 1,
          "htmlURL": "http://the-url.com",
          "id": 1,
          "language": "en",
          "name": "name",
          "openIssuesNumber": 1,
          "owner": Object {
            "avatarURL": "http://the-url.com",
            "htmlURL": "http://the-url.com",
            "id": 1,
            "username": "username",
          },
          "starsNumber": 1,
          "updatedAt": "05/05/2005",
        }
      }
    />
  </div>
  <div
    className={
      Object {
        "has": [Function],
        "is": [Function],
        "mix": [Function],
        "split": [Function],
        "state": [Function],
        "toString": [Function],
      }
    }
    key="1"
  >
    <RepositoryPreview
      onOwnerClick={[Function]}
      repository={
        Object {
          "description": "description",
          "forksNumber": 1,
          "htmlURL": "http://the-url.com",
          "id": 1,
          "language": "en",
          "name": "name",
          "openIssuesNumber": 1,
          "owner": Object {
            "avatarURL": "http://the-url.com",
            "htmlURL": "http://the-url.com",
            "id": 1,
            "username": "username",
          },
          "starsNumber": 1,
          "updatedAt": "05/05/2005",
        }
      }
    />
  </div>
  <div
    className={
      Object {
        "has": [Function],
        "is": [Function],
        "mix": [Function],
        "split": [Function],
        "state": [Function],
        "toString": [Function],
      }
    }
    key="1"
  >
    <RepositoryPreview
      onOwnerClick={[Function]}
      repository={
        Object {
          "description": "description",
          "forksNumber": 1,
          "htmlURL": "http://the-url.com",
          "id": 1,
          "language": "en",
          "name": "name",
          "openIssuesNumber": 1,
          "owner": Object {
            "avatarURL": "http://the-url.com",
            "htmlURL": "http://the-url.com",
            "id": 1,
            "username": "username",
          },
          "starsNumber": 1,
          "updatedAt": "05/05/2005",
        }
      }
    />
  </div>
  <div
    className={
      Object {
        "has": [Function],
        "is": [Function],
        "mix": [Function],
        "split": [Function],
        "state": [Function],
        "toString": [Function],
      }
    }
    key="1"
  >
    <RepositoryPreview
      onOwnerClick={[Function]}
      repository={
        Object {
          "description": "description",
          "forksNumber": 1,
          "htmlURL": "http://the-url.com",
          "id": 1,
          "language": "en",
          "name": "name",
          "openIssuesNumber": 1,
          "owner": Object {
            "avatarURL": "http://the-url.com",
            "htmlURL": "http://the-url.com",
            "id": 1,
            "username": "username",
          },
          "starsNumber": 1,
          "updatedAt": "05/05/2005",
        }
      }
    />
  </div>
  <div
    className={
      Object {
        "has": [Function],
        "is": [Function],
        "mix": [Function],
        "split": [Function],
        "state": [Function],
        "toString": [Function],
      }
    }
    key="1"
  >
    <RepositoryPreview
      onOwnerClick={[Function]}
      repository={
        Object {
          "description": "description",
          "forksNumber": 1,
          "htmlURL": "http://the-url.com",
          "id": 1,
          "language": "en",
          "name": "name",
          "openIssuesNumber": 1,
          "owner": Object {
            "avatarURL": "http://the-url.com",
            "htmlURL": "http://the-url.com",
            "id": 1,
            "username": "username",
          },
          "starsNumber": 1,
          "updatedAt": "05/05/2005",
        }
      }
    />
  </div>
  <div
    className={
      Object {
        "has": [Function],
        "is": [Function],
        "mix": [Function],
        "split": [Function],
        "state": [Function],
        "toString": [Function],
      }
    }
    key="1"
  >
    <RepositoryPreview
      onOwnerClick={[Function]}
      repository={
        Object {
          "description": "description",
          "forksNumber": 1,
          "htmlURL": "http://the-url.com",
          "id": 1,
          "language": "en",
          "name": "name",
          "openIssuesNumber": 1,
          "owner": Object {
            "avatarURL": "http://the-url.com",
            "htmlURL": "http://the-url.com",
            "id": 1,
            "username": "username",
          },
          "starsNumber": 1,
          "updatedAt": "05/05/2005",
        }
      }
    />
  </div>
  <div
    className={
      Object {
        "has": [Function],
        "is": [Function],
        "mix": [Function],
        "split": [Function],
        "state": [Function],
        "toString": [Function],
      }
    }
    key="1"
  >
    <RepositoryPreview
      onOwnerClick={[Function]}
      repository={
        Object {
          "description": "description",
          "forksNumber": 1,
          "htmlURL": "http://the-url.com",
          "id": 1,
          "language": "en",
          "name": "name",
          "openIssuesNumber": 1,
          "owner": Object {
            "avatarURL": "http://the-url.com",
            "htmlURL": "http://the-url.com",
            "id": 1,
            "username": "username",
          },
          "starsNumber": 1,
          "updatedAt": "05/05/2005",
        }
      }
    />
  </div>
  <div
    className={
      Object {
        "has": [Function],
        "is": [Function],
        "mix": [Function],
        "split": [Function],
        "state": [Function],
        "toString": [Function],
      }
    }
  >
    <PaginationControls
      currentPage={1}
      onPageRequest={[Function]}
      totalPages={1}
    />
  </div>
</div>
`;
